numpy/lib/format.py

```diff
@@ -259,6 +259,33 @@ def dtype_to_descr(dtype):
     else:
         return dtype.str
 
+def descr_to_dtype(descr):
```

Comment:
Could do with a short docstring explaining how this is different from `np.dtype(descr)`

Reply:
added